src/main/java/org/mifos/connector/ams/paygops/camel/route/PaygopsRouteBuilder.java

```diff
@@ -66,20 +68,32 @@ public void configure() {
                 .log(LoggingLevel.INFO, "Paygops Validation Response Received")
                 .process(exchange -> {
                     // processing success case
-                    String body = exchange.getIn().getBody(String.class);
-                    ObjectMapper mapper = new ObjectMapper();
-                    PaygopsResponseDto result = mapper.readValue(body, PaygopsResponseDto.class);
-                    logger.info("body : "+ result);
-                    //JSONObject jsonObject = new JSONObject(body);
-                    if (result.getReconciled()){
+                    try {
+                        String body = exchange.getIn().getBody(String.class);
+                        ObjectMapper mapper = new ObjectMapper();
+                        PaygopsResponseDto result = mapper.readValue(body, PaygopsResponseDto.class);
+                        logger.info("body : " + result);
+                        //JSONObject jsonObject = new JSONObject(body);
+                        if (result.getReconciled()) {
                             logger.info("Paygops Validation Successful");
                             exchange.setProperty(PARTY_LOOKUP_FAILED, false);
+                        } else {
+                            logger.info("Paygops Validation Unsuccessful, Reconciled field returned false");
+                            String errorInfo = "Reconciled field returned false";
+                            String errorDesc = result.toString();
+                            setErrorCamelInfo(exchange,errorDesc,123,errorInfo);
+
+                            exchange.setProperty(PARTY_LOOKUP_FAILED, true);
                         }
-                        else {
-                        logger.info("Paygops Validation Unsuccessful, Reconciled field returned false");
+                    } catch (Exception e) {
+                        logger.info("Body data could not be parsed, setting validation as failed");
```

Comment:
Line 89,90,91 also goes inside the function to keep caller function clean.

Reply:
@ankita10r : Did you get a chance to look at these review comments?

Reply:
Sorry I had missed them, addressing them now